refactor: migrate Makefile to Taskfile

GabriFedi97 opened the pull request:
Add Taskfile including all the commands currently available through the Makefile.
Updated the documentation accordingly (BUILD.md).

Closes: #54

GabriFedi97 committed:
refactor: migrate Makefile to Taskfile
Migrated all the commands available in the Makefile
to the Taskfile. Task is now required to execute
pre-defined commands.

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

GabriFedi97 committed:
ci(dagger): add method to get all target extensions
Add a command that more generally retrieve all the
extension folders by checking which one contains
the metadata.hcl file.

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

GabriFedi97 committed:
fix: run prereqs once

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

mbiagetti commented on BUILD.md:
this part is not needed anymore, right?

mbiagetti commented:
Shall we replace here with `Taskfile.yml` instead of `Makefile`?

 https://github.com/GabriFedi97/postgres-extensions-containers/blob/82863aa42a48624416e3f6a551afc67ecbb8da90/.github/workflows/bake.yml#L42

mbiagetti committed:
chore: review (#2)

Signed-off-by: Matteo <matteo.biagetti@enterprisedb.com>

mbiagetti reviewed (approved):
LGTM!

gbartolini committed:
docs: add link to dagger and task

Signed-off-by: Gabriele Bartolini <gabriele.bartolini@enterprisedb.com>

NiccoloFei committed:
review: remove unused command

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

NiccoloFei committed:
docs: review

Signed-off-by: Niccolò Fei <niccolo.fei@enterprisedb.com>

GabriFedi97 committed:
fix: precondition check for update-os-libs
In dagger module we check whether the specified target is valid and
has the auto_update_os_libs enabled in metadata before attempting
to execute the update.
We exit with a proper error otherwise.

Signed-off-by: Gabriele Fedi <gabriele.fedi@enterprisedb.com>

NiccoloFei reviewed (approved):
LGTM
